rust-connectors/sources/postgres/src/connect.rs

```diff
@@ -0,0 +1,211 @@
+use crate::convert::convert_replication_event;
+use crate::{Error, PgConnectorOpt};
+use fluvio::metadata::topic::TopicSpec;
+use fluvio::{Fluvio, Offset, TopicProducer};
+use fluvio_model_postgres::{Column, LogicalReplicationMessage, ReplicationEvent};
+use once_cell::sync::Lazy;
+use postgres_protocol::message::backend::{
+    LogicalReplicationMessage as PgReplication, ReplicationMessage,
+};
+use std::collections::BTreeMap;
+use std::pin::Pin;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use tokio_postgres::config::ReplicationMode;
+use tokio_postgres::replication::LogicalReplicationStream;
+use tokio_postgres::types::PgLsn;
+use tokio_postgres::{Client, NoTls};
+use tokio_stream::StreamExt;
+
+const TIME_SEC_CONVERSION: u64 = 946_684_800;
+static EPOCH: Lazy<SystemTime> =
+    Lazy::new(|| UNIX_EPOCH + Duration::from_secs(TIME_SEC_CONVERSION));
+
+/// A Fluvio connector for Postgres CDC.
+pub struct PgConnector {
+    /// The connector configuration.
+    config: PgConnectorOpt,
+    /// The Postgres client for streaming replication changes.
+    pg_client: Client,
+    /// The Fluvio producer for recording change events.
+    producer: TopicProducer,
+    /// The current Log Sequence Number (offset) to read in the replication stream.
+    lsn: Option<PgLsn>,
+    /// Caches the schema for each new table we see, grouped by relation_id
+    relations: BTreeMap<u32, Vec<Column>>,
+}
+
+impl PgConnector {
+    pub async fn new(config: PgConnectorOpt) -> eyre::Result<Self> {
+        tracing::info!("Initializing PgConnector");
+        let fluvio = Fluvio::connect().await?;
+        tracing::info!("Connected to Fluvio");
+        if !config.skip_setup {
+            let _ = Self::create_replication_slot(&config).await?;
+        }
+
+        let admin = fluvio.admin().await;
+
+
+        let topics = admin.list::<TopicSpec, _>(vec![]).await?;
+        let topic_exists = topics.iter().any(|t| t.name == config.common.fluvio_topic);
+        if !topic_exists {
+            return Err(Error::TopicNotFound(config.common.fluvio_topic.to_string()).into());
+        }
+
+        let consumer = fluvio
```

Comment:
could you move this to fluvio-connector-common? I mean just like we have create_producer() there

Reply:
Or I can just do that in a separate PR

Reply:
Oh, right. Good catch.